src/sagemaker_sklearn_extension/contrib/taei/latent_space_oversampler.py

```diff
@@ -0,0 +1,106 @@
+import torch
+import numpy as np
+from sklearn.model_selection import train_test_split
+from sklearn.utils import check_X_y
+
+
+class LatentSpaceOversampler:
```

Comment:
There's a [`BaseSampler`](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/331c6033e0abb5fb23a106fcf2144c4416455672/imblearn/base.py#L116) and [`BaseOverSampler`](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/331c6033e0abb5fb23a106fcf2144c4416455672/imblearn/over_sampling/base.py#L11) class from imbalanced learn which inherits form a `SampleMixin` class. Do you see value in inheriting from them?

Reply:
No. And that would require taking dependency on imbalanced-learn which we currently don't have